src/watchers/file.ts

```diff
@@ -0,0 +1,84 @@
+import * as fs from "fs";
+import {Watcher, RawEAVC} from "./watcher";
+import {ID} from "../runtime/runtime";
+
+class FileWatcher extends Watcher {
+
+    setup() {
+        let {program:me} = this;
+        me.load(`
+Attach errors to the associated file
+~~~
+search
+  error = [#file/error file code message]
+
+bind
+  file.error += error
+~~~     
+        `)
+        if(fs.readFile) {
+            me.watch("read a file", ({find, record}) => {
+                let file = find("file/read");
+                return [
+                    record({file, path: file.path, encoding: file.encoding})
+                ]
+            })
+
+            me.asObjects<{file:ID, path:string, encoding:string}>(({adds, removes}) => {
+                Object.keys(adds).forEach((id) => {
+                    let {file, path, encoding} = adds[id];
+                    fs.readFile(path, {encoding: encoding}, function(err, contents){
+                        if (!err) {
+                            let changes:RawEAVC[] = [];
+                            changes.push(
+                                [file, "contents", contents, 1],
+                            );
+                            me.inputEAVs(changes);
```

Comment:
This whole branch can be reduced to `me.inputEAVs([[file, "contents", contents]])`. The count is optional and defaults to 1.